app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -257,13 +257,15 @@ def index_for_group_type
   def create
     user_id = params.require(:userId)
     group_id = params[:groupId] || UserGroup.find_by(group_type: params.require(:groupType)).id
-    group = UserGroup.find(group_id)
 
     if group_id.is_a?(String) && group_id.include?("_") # Temporary hack to support some old system roles, will be removed once all roles are
       # migrated to the new system.
       group_type = group_id_of_old_system_to_group_type(group_id)
       original_group_id = group_id.split("_").last
-      if group_type == UserGroup.group_types[:councils]
+      unless current_user.can_edit_team?(original_group_id)
+        return head :unauthorized
+      end
```

Comment:
Cool! Now that we have established that this works, you can take it a step further with
`return ... unless ...` on one single line :smile:

Reply:
Done for all the three.